src/services/review/RepoCheckerService.ts

```diff
@@ -316,54 +410,170 @@ export default class RepoCheckerService {
       fs.mkdirSync(folderPath, { recursive: true })
     }
-    const filePath = path.join(SITE_CHECKER_REPO_PATH, repo, "log.csv")
+    const filePath = path.join(SITE_CHECKER_REPO_PATH, repo, REPO_LOG)
     const stringifiedErrors = this.convertToCSV(errors)
     if (!fs.existsSync(filePath)) {
       fs.appendFileSync(filePath, stringifiedErrors)
     } else {
       fs.writeFileSync(filePath, stringifiedErrors)
     }
-
+    // pushing since there is no real time requirement for user
     return ResultAsync.fromPromise(
       this.git
         .cwd({ path: SITE_CHECKER_REPO_PATH, root: false })
+        .fetch()
+        .merge(["origin/main"])
         .add(["."])
         .commit("site checker logs")
         .push(),
       (error) => new SiteCheckerError(`${error}`)
-    )
+    ).map(() => errors)
   }
 
-  checkRepo(repo: string): ResultAsync<undefined, SiteCheckerError> {
+  // adapted from https://stackoverflow.com/questions/56427009/how-to-return-papa-parsed-csv-via-promise-async-await
+  generateRepoErrorsFromCsv = (repo: string): Promise<RepoError[]> => {
+    const filePath = path.join(SITE_CHECKER_REPO_PATH, repo, REPO_LOG)
+    const file = fs.createReadStream(filePath)
+    return new Promise((resolve, reject) => {
+      Papa.parse(file, {
+        header: true,
+        complete(results) {
+          // validate the csv
+          if (!results.data.every(isRepoError)) {
+            reject(new Error("Invalid CSV"))
+          }
+          resolve(results.data as RepoError[])
+        },
+        error(error) {
+          reject(error)
+        },
+      })
+    })
+  }
+
+  readRepoErrors = (
+    repo: string
+  ): ResultAsync<BrokenLinkErrorDto, SiteCheckerError> => {
+    const filePath = path.join(SITE_CHECKER_REPO_PATH, repo, REPO_LOG)
+    if (!fs.existsSync(filePath)) {
+      logger.error(`SiteCheckerError: Folder does not exist for ${repo}`)
+      return errAsync(new SiteCheckerError(`Folder does not exist for ${repo}`))
+    }
+
+    return this.isCurrentlyErrored(repo).andThen((isError) => {
+      if (isError) {
+        return errAsync(new SiteCheckerError(`Error checking repo ${repo}`))
+      }
+
+      return this.isCurrentlyLocked(repo)
+        .andThen(() => okAsync<BrokenLinkErrorDto>({ status: "loading" }))
+        .orElse(() =>
+          ResultAsync.fromPromise(
+            this.generateRepoErrorsFromCsv(repo),
+            (error) => {
+              logger.error(
+                `SiteCheckerError: Error reading csv for repo ${repo}: ${error}`
+              )
+              return new SiteCheckerError(
+                `Error reading csv for repo ${repo}: ${error}`
+              )
+            }
+          ).andThen((errors) =>
+            okAsync<BrokenLinkErrorDto>({
+              errors,
+              status: "success",
+            })
+          )
+        )
+    })
+  }
+
+  checkRepo(repo: string): ResultAsync<BrokenLinkErrorDto, SiteCheckerError> {
```

Comment:
checkRepo is mapped to the POST call made by the user clicking on the run site check?

Reply:
yes, this is the call that updates our internal state